packages/viewer/src/components/scene-tree-table/scene-tree-table.tsx

```diff
@@ -0,0 +1,511 @@
+   * while scrolling.
+   *
+   * This prop will be automatically populated based on the `overScanCount` prop
+   * specified in the parent `<vertex-scene-tree />` element.
+   */
+  @Prop()
+  public overScanCount = 25;
+
+  /**
+   * A callback that is invoked immediately before a row is about to rendered.
+   * This callback can return additional data that can be bound to in a
+   * template.
+   *
+   * This prop will be automatically populated based on the `rowData` prop
+   * specified in the parent `<vertex-scene-tree />` element.
+   *
+   * @example
+   *
+   * ```html
+   * <script>
+   *   const table = document.querySelector('vertex-scene-tree-table');
+   *   table.rowData = (row) => {
+   *     return { func: () => console.log('row', row.node.name) };
+   *   }
+   * </script>
+   *
+   * <vertex-scene-tree>
+   *  <vertex-scene-tree-table>
+   *    <vertex-scene-tree-table-column>
+   *      <template>
+   *        <button event:click="{{row.data.func}}">Hi</button>
+   *      </template>
+   *    </vertex-scene-tree-table-column>
+   *  </vertex-scene-tree-table>
+   * </vertex-scene-tree>
+   * ```
+   */
+  @Prop()
+  public rowData?: RowDataProvider;
+
+  /**
+   * @internal
+   */
+  @Prop({ mutable: true })
+  public layoutOffset = 0;
+
+  /**
+   * @internal
+   */
+  @Prop({ mutable: true })
+  public scrollOffset = 0;
+
+  /**
+   * @internal
+   */
+  @Prop({ mutable: true })
+  public layoutHeight?: number;
+
+  /**
+   * A flag that disables the default interactions of this component. If
+   * disabled, you can use the event handlers to be notified when certain
+   * operations are performed by the user.
+   */
+  @Prop()
+  public interactionsDisabled = false;
+
+  /**
+   * A flag that disables selection of the node's parent if the user selects
+   * the row multiple times. When enabled, selection of the same row multiple
+   * times will recursively select the next unselected parent until the root
+   * node is selected.
+   */
+  @Prop()
+  public recurseParentSelectionDisabled = false;
+
+  @Element()
+  private hostEl!: HTMLElement;
+
+  @State()
+  private columnGridLayout = '1fr';
+
+  @State()
+  private hoveredNodeId?: string;
+
+  @State()
+  private isComputingCellHeight = true;
+
+  /**
+   * This stores internal state that you want to preserve across live-reloads,
+   * but shouldn't trigger a refresh if the data changes. Marking this with
+   * @State to allow to preserve state across live-reloads.
+   */
+  @State()
+  private stateMap: StateMap = {
+    startIndex: 0,
+    endIndex: 0,
+    viewportRows: [],
+  };
+
+  private lastStartIndex = 0;
+  private resizeObserver?: ResizeObserver;
+
+  private tableElement?: HTMLDivElement;
+  private headerElement?: HTMLDivElement;
+  private columnElements: HTMLVertexSceneTreeTableColumnElement[] = [];
+
+  public componentWillLoad(): void {
+    this.updateColumnElements();
+    this.computeColumnGridLayout();
+    this.createPools();
```

Comment:
I think we'll also need to call some variation of these methods if the column counts change. Can probably handle that in a follow up.